src/main/java/com/puppycrawl/tools/checkstyle/checks/OrderedPropertiesCheck.java

```diff
@@ -0,0 +1,202 @@
+     */
+    public OrderedPropertiesCheck() {
+        setFileExtensions("properties");
+    }
+
+    @Override
+    protected void processFiltered(File file, FileText fileText) throws CheckstyleException {
+        final SequencedProperties properties = new SequencedProperties();
+        try (InputStream inputStream = Files.newInputStream(file.toPath())) {
+            properties.load(inputStream);
+        }
+        catch (IOException ex) {
+            throw new CheckstyleException(UNABLE_TO_OPEN_FILE + file.getPath(), ex);
+        }
+
+        String previousProp = "";
+
+        final Enumeration<Object> keys = properties.keys();
+
+        while (keys.hasMoreElements()) {
+
+            final String propKey = (String) keys.nextElement();
+
+            //if (previousProp.compareTo(propKey) > 0) {
+            if (String.CASE_INSENSITIVE_ORDER.compare(previousProp, propKey) > 0) {
+
+                final int lineNumber = getLineNumber(fileText, propKey);
+                log(lineNumber, MSG_KEY, propKey, previousProp);
+            }
+
+            previousProp = propKey;
+        }
+    }
+
+    /**
+     * Method returns the line number where the key is detected.
+     *
+     * @param fileText {@link FileText} object contains the lines to process
+     * @param keyName  key name to look for
+     * @return line number of first occurrence. If no key found in properties file, 0 is returned
+     */
+    private static int getLineNumber(FileText fileText, String keyName) {
+        final Pattern keyPattern = getKeyPattern(keyName);
+        int humanIndex = 0;
+
+        for (int index = 0; index < fileText.size(); index++) {
+            final String line = fileText.get(index);
+            if (keyPattern.matcher(line).matches()) {
+                // human line index
+                humanIndex = index + 1;
+                break;
+            }
+        }
+        // if no key is found return 0
+        return humanIndex;
+    }
+
+    /**
+     * Method returns regular expression pattern given key name.
+     *
+     * @param keyName
+     *            key name to look for
+     * @return regular expression pattern given key name
+     */
+    private static Pattern getKeyPattern(String keyName) {
+        final String keyPatternString = "^" + SPACE_PATTERN.matcher(keyName)
+                .replaceAll(Matcher.quoteReplacement("\\\\ ")) + "[\\s:=].*$";
+        return Pattern.compile(keyPatternString);
+    }
+
+    /**
+     * Private property implementation that keeps order of properties like in file.
+     *
+     * @noinspection ClassExtendsConcreteCollection, SerializableHasSerializationMethods
+     */
+    private static class SequencedProperties extends Properties {
+
+        private static final long serialVersionUID = 1L;
+
+        /**
+         * Holding the keys in the same order than in the file.
+         */
+        private final List<Object> keyList = new ArrayList<>();
```

Comment:
> 173: keyList Non-serializable field 'keyList' in a Serializable class 

This inspection may have to be suppressed.

Reply:
I was setting it to transient. It does not need to be serialized.